File: app/src/main/java/cc/lixiaoyu/wanandroid/util/SpUtil.java
package cc.lixiaoyu.wanandroid.util;

import android.content.Context;
import android.content.SharedPreferences;

import cc.lixiaoyu.wanandroid.app.MyApplication;

/**
 * SharedPreperence的工具类
 */
public class SpUtil {
    private static Context context = MyApplication.getGlobalContext();
    public static final String FILE_DEFAULT = "WanAndroid";

    public static void saveData(String key, Object data) {

        SharedPreferences sharedPreferences = context
                .getSharedPreferences(FILE_DEFAULT, Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();

        if (data instanceof Integer) {
            editor.putInt(key, (Integer) data);
        } else if (data instanceof Boolean) {
            editor.putBoolean(key, (Boolean) data);
        } else if (data instanceof String) {
            editor.putString(key, (String) data);
        } else if (data instanceof Float) {
            editor.putFloat(key, (Float) data);
        } else if (data instanceof Long) {
            editor.putLong(key, (Long) data);
        }

        editor.commit();
    }

    public static Object getData(String key, Object defaultObject) {


        SharedPreferences sharedPreferences = context.getSharedPreferences
                (FILE_DEFAULT, Context.MODE_PRIVATE);

        //defValue为为默认值，如果当前获取不到数据就返回它
        if (defaultObject instanceof Integer) {
            return sharedPreferences.getInt(key, (Integer) defaultObject);
        } else if (defaultObject instanceof Boolean) {
            return sharedPreferences.getBoolean(key, (Boolean) defaultObject);
        } else if (defaultObject instanceof String) {
            return sharedPreferences.getString(key, (String) defaultObject);
        } else if (defaultObject instanceof Float) {
            return sharedPreferences.getFloat(key, (Float) defaultObject);
        } else if (defaultObject instanceof Long) {
            return sharedPreferences.getLong(key, (Long) defaultObject);
        }

        return null;
    }
}

File: app/src/main/java/cc/lixiaoyu/wanandroid/util/storage/SPUtil.kt
package cc.lixiaoyu.wanandroid.util.storage

import android.annotation.SuppressLint
import android.content.Context
import cc.lixiaoyu.wanandroid.app.WanApplication.Companion.globalContext
import java.lang.IllegalArgumentException

/**
 * SharedPreperence的工具类
 */
@SuppressLint("StaticFieldLeak")
object SPUtil {
    private val context = globalContext
    private const val FILE_DEFAULT = "WanAndroid"
    
    @JvmStatic
    fun saveData(key: String?, data: Any) {
        val sharedPreferences = context
            ?.getSharedPreferences(FILE_DEFAULT, Context.MODE_PRIVATE) ?: return
        val editor = sharedPreferences.edit()
        when (data) {
            is Int -> {
                editor.putInt(key, data)
            }
            is Boolean -> {
                editor.putBoolean(key, data)
            }
            is String -> {
                editor.putString(key, data)
            }
            is Float -> {
                editor.putFloat(key, data)
            }
            is Long -> {
                editor.putLong(key, data)
            }
            else -> {
                throw IllegalArgumentException("SpUtil doesn't support this type")
            }
        }
        editor.apply()
    }

    @JvmStatic
    fun getData(key: String?, defaultObject: Any): Any? {
        val sharedPreferences = context?.getSharedPreferences(FILE_DEFAULT, Context.MODE_PRIVATE) ?: return null

        //defValue为为默认值，如果当前获取不到数据就返回它
        when (defaultObject) {
            is Int -> {
                return sharedPreferences.getInt(key, defaultObject)
            }
            is Boolean -> {
                return sharedPreferences.getBoolean(key, defaultObject)
            }
            is String -> {
                return sharedPreferences.getString(key, defaultObject)
            }
            is Float -> {
                return sharedPreferences.getFloat(key, defaultObject)
            }
            is Long -> {
                return sharedPreferences.getLong(key, defaultObject)
            }
            else -> {
                throw IllegalArgumentException("SpUtil doesn't support this type")
            }
        }
    }
}